ext/linux/osq_lock.h

```diff
@@ -321,19 +386,31 @@ void osq_unlock(uint64_t *osq, unsigned long cpu_number)
         return;
     }
 
+    /*
+     * Wait for another stable next, or get NULL if the queue is empty.
+     */
     next = osq_wait_next(lock, node, NULL, cpu_number);
     if (next)
         WRITE_ONCE(next->locked, 1);
 }
 
+
+/* standard lockhammer lock_acquire and lock_release interfaces */
 unsigned long __attribute__((noinline)) lock_acquire (uint64_t *lock, unsigned long threadnum)
 {
-    /* TODO: need to implement mutex slow path like
-     * __mutex_lock_common() and mutex_optimistic_spin()
-     * in linux/kernel/locking/mutex.c
+    /*
+     * TODO: need to implement mutex slow path like __mutex_lock_common()
```

Comment:
Is this actually a TODO, or just a reminder to people that this is not implemented in userspace?

Reply:
It would be good to make this distinction and document it. At the top of osq_lock.h, add a comment block like what Lucas had done for TBB, JVM Object monitor etc.